### ISSUE TITLE
Create a GitHub Action to clean templated repos on create

### ISSUE THREAD

**Issue by JacobCoffee** [JacobCoffee/starlite-barebones-template#4]
Currently there is no `.gitignore` version for GitHub templates.
A workaround is to have a `GitHub Action` do some cleanup `on: create` when someone uses the template.

Cleanup includes (may not be 100% inclusive):

- [ ] `README.md` and app/* project name changes (via `sed`❓)
- [ ] Version wiping (via `bump2version`❓)
- [ ] Documenation fixes

Example (from https://github.com/orgs/community/discussions/22183)
```yml
name: first-time-setup
run-name: first time setup of repo

on:
  # run when branch created (repo generated from template)
  create:

# only keep latest run of this workflow
concurrency:
  group: first-time-setup
  cancel-in-progress: true

jobs:
  first-time-setup:
    # ensure run only once, when repo generated
    if: github.run_number == 1

    runs-on: ubuntu-latest

    steps:
      # get main branch repo contents
      - name: Checkout code
        uses: actions/checkout@v3

      # remove files not needed for user instance of template
      - name: Remove unneeded files
        run: rm -f README.md demo.md testbed.md

      # personalize readme for user
      - name: Init readme
        run: |
          echo "
   
          Personalized readme text. Hello ${{ github.repository_owner }}!

          " > README.md

      # save changed files
      - name: Commit changed files
        uses: stefanzweifel/git-auto-commit-action@v4
        with:
          commit_message: "Setup repo"
```

**Comment by JacobCoffee** [comment]
> README.md and app/* project name changes (via sedquestion)

on this, i think this is more into cookiecutter territory?
